api-authorization/server/lib/authorization-middleware.ts

```diff
@@ -0,0 +1,39 @@
+import { NextFunction, Request, Response } from 'express';
+import jwt from 'jsonwebtoken';
+import { ClientError } from './client-error.js';
+
+const hashKey = process.env.TOKEN_SECRET;
+if (!hashKey) throw new Error('TOKEN_SECRET not found in .env');
+
+type AuthenticatedUser = {
+  userId: number;
+  username: string;
+  iat: number;
+};
+
+export function authMiddleware(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  const auth = req.headers.authorization;
+  const token = auth && auth.split('Bearer ')[1];
+  if (!token) throw new ClientError(401, 'authentication required');
+  req.user = jwt.verify(token, hashKey ?? '') as AuthenticatedUser;
```

Comment:
The file `auth.d.ts` defines what the User should look like. You can take advantage of that definition by using `as Request['user']` instead of declaring your own AuthenticatedUser type.